kong/controlplane-connector/components/kong.gw.manager/src/main/java/org/wso2/kong/client/KongFederatedAPIDiscovery.java

```diff
@@ -105,214 +109,219 @@ public void init(Environment environment, String organization)
-            if (selectedAPILevelRateLimitPolicy != null) {
-                api.setApiLevelPolicy(selectedAPILevelRateLimitPolicy);
-            }
-            retrievedAPIs.add(api);
-        }
 
-        // If there are Services without APIs, we can still retrieve them as APIs
-        PagedResponse<KongService> servicesResp = apiGatewayClient.listServices(controlPlaneId, 1000);
-        List<KongService> services;
-        if (servicesResp != null && servicesResp.getData() != null) {
-            services = servicesResp.getData();
-        } else {
-            services = java.util.Collections.emptyList();
-        }
-
-        for (KongService svc : services) {
-            // Skip if this service is already linked to an API
-            if (linkedServices.contains(svc.getId())) {
-                continue;
-            }
-            PagedResponse<KongRoute> resp = apiGatewayClient.listRoutesByServiceId(controlPlaneId, svc.getId(), 1000);
-            List<KongRoute> routes = (resp != null && resp.getData() != null) ?
-                    resp.getData() : java.util.Collections.emptyList();
-
-            PagedResponse<KongPlugin> pluginsResp = apiGatewayClient.listPluginsByServiceId(
-                    controlPlaneId, svc.getId(), 100);
-            List<KongPlugin> plugins = (pluginsResp != null && pluginsResp.getData() != null)
-                ? pluginsResp.getData() : java.util.Collections.<KongPlugin>emptyList();
-
-            APIIdentifier apiId = new APIIdentifier("admin", svc.getName(), "v1");
-            API api = new API(apiId);
-            api.setDisplayName(svc.getName());
-            api.setContext(svc.getName());
-            api.setContextTemplate(svc.getName().toLowerCase().replace(" ", "-"));
-            api.setUuid(svc.getId());
-            api.setDescription("");
-            api.setOrganization(organization);
-            api.setRevision(false);
-
-            if (svc.getUpdatedAt() != null) {
-                api.setLastUpdated(Date.from(java.time.Instant.ofEpochSecond(svc.getUpdatedAt())));
-            }
-            if (svc.getCreatedAt() != null) {
-                api.setCreatedTime(Long.toString(svc.getCreatedAt()));
+            // If there are Services without APIs, we can still retrieve them as APIs
+            PagedResponse<KongService> servicesResp = apiGatewayClient.listServices(controlPlaneId, 1000);
+            List<KongService> services;
+            if (servicesResp != null && servicesResp.getData() != null) {
+                services = servicesResp.getData();
+            } else {
+                services = java.util.Collections.emptyList();
             }
 
-            api.setInitiatedFromGateway(true);
-            api.setGatewayVendor("external");
-            api.setGatewayType(environment.getGatewayType());
+            for (KongService svc : services) {
+                // Skip if this service is already linked to an API
+                if (linkedServices.contains(svc.getId())) {
```

Comment:
Shall we use a hashset instead of arraylist? If all service IDs are unique then it's better to go with a hashset to reduce the lookup time.